text/0038-scrubbing-sensitive-data.md

```diff
@@ -0,0 +1,135 @@
+  - `maybe` means that the data scrubbing Regexes are not run by default. If the user has advanced data scrubbing set (in Sentry.io under: Project Settings > Security & Privacy > Advanced Data Scrubbing) those custom rules are applied to the field and if a rule matches the whole content of the field is removed.
+  - `no` there is no possibility of data scrubbing for this field.
+- The regexes for data scrubbing are defined [here](https://github.com/getsentry/relay/blob/master/relay-general/src/pii/regexes.rs#L103-L274)
+- Some regexes can just remove the sensitive part of the content (like IP and SSH keys regexes).
+- Some of the regexes (like password regex) will remove the complete content of a field. This is because it is unstructured data. Relay does not know if the content is a SQL query, a JSON object, an URL, a Elasticsearch/MongoDB/whatever query in JSON format, or something else.
+
+### Currently perferred solution:
+
+A combination of some of the options listed below:
+
+- Have the SDKs store structured data (Option B)
+- If Relay does not get any structured data from Option B, Relay does Option C.
+- If Relay can not figure out what kind of data is in the field, do matching like we do now.
+
+### Option A): Remove Sensitive Data in SDKs
+
+At the time when SDKs set the data in an event, it knows what the data represents and can remove sensitive information. This way Relay has to beliefe that the SDK does the right thing and does not need to scrub data.
+
+_Pros:_
+
+-
+
+_Cons:_
+
+-
+
+### Option B): Store data in a structured way in SDKs
+
+The content should not be a simple string but structured data. The content could be a template string where all the sensitive values are removed plus a dictionary with the values to insert into the string.
+Example `span.description` should be a string with named parameters in the format `select * from user where email=%(email)s;` or `POST /api/v1/update_password?new_password=%(new_password)s` and in `span.data` there should be `{"email": "test@example.com"}` or `{ "new_password": "123456" }` respectively.
+Same goes for `breadcrumb.message/breadcrumb.data`, `logentry.message/logentry.params`, and `message.message/message.params`.
+
+We need to identify all the fields we need to do this.
+
+We should model this Option after how `Logentry` does this now.
+
+Note: If we change the span.description then the hash for existing performance issues will be changed and existing performance issues will be recreated, so users would have duplicate performance issues in their list of issues. (Which we can just document in the CHANGELOG and everything should be fine.)
+
+_Pros:_
+
+- Relay would not have to reverse-engineer the semantics of the information supplied by the SDK.
+
+_Cons:_
+
+- Could be complex for nested JSON objects or for "Array of objects" kind of data.
+
+### Option C): Relay identifies what kind of data is present and parses it.
```

Comment:
I like this option because we don't run into this issue https://github.com/getsentry/rfcs/pull/38/files#r1029153081